Add auto-generated docs for custom math examples

sjpfenninger opened the pull request:
Summary of changes in this pull request:

* Changed comment strings into commented-out YAML with a "title" and "description" for each example (the `# ---` comment marks the end of the "docs comment")
* Added code to auto-generate custom math documentation

Reviewer checklist:

- [ ] Test(s) added to cover contribution
- [ ] Documentation updated
- [ ] Changelog updated
- [ ] Coverage maintained or improved

brynpickering commented on doc/helpers/generate_custom_constraints.py:
```diff
@@ -0,0 +1,80 @@
+
+
+def process_file(path):
+    with open(path, "r") as f:
```
`path.read_text()` handles this without the need for the context manager.

brynpickering commented on doc/helpers/generate_custom_constraints.py:
```diff
@@ -0,0 +1,80 @@
+# Licensed under the Apache 2.0 License (see LICENSE file).
+
+"""
+generate_custom_constraints.py
```
`generate_custom_math_examples`?

brynpickering commented on doc/helpers/generate_custom_constraints.py:
```diff
@@ -0,0 +1,80 @@
+generate_custom_constraints.py
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Generates an `.rst` file listing all custom constraints.
```
```suggestion
Generates an `.rst` file listing all custom math examples.
```

brynpickering commented on doc/_static/custom_math/uptime_downtime_limits.yaml:
```diff
@@ -1,16 +1,22 @@
-# capacity_factor_max
-# downtime_periods (from CSV as a timeseries)
-# uptime_limit
+# description: |
```
the pipe leave the result with the line breaks when loaded in, `>` ignores the line breaks. Do we have a preference?

sjpfenninger replied:
The pipe is needed to keep formatting like paragraphs intact, so I think we should always go with the pipe.

sjpfenninger commented:
Closing this because it is superseded by #526 (which addresses all of the issues pointed out in the review, with the possible exception of the question on `|` vs `>`)
